secret/secrets_manager_client.go

```diff
@@ -123,7 +125,56 @@ func (c *BasicSecretsManagerClient) CreateSecret(ctx context.Context, in *secret
 
 // GetSecretValue gets the decrypted value of an existing secret.
 func (c *BasicSecretsManagerClient) GetSecretValue(ctx context.Context, in *secretsmanager.GetSecretValueInput) (*secretsmanager.GetSecretValueOutput, error) {
-	return nil, errors.New("TODO: implement")
+	if err := c.setup(); err != nil {
+		return nil, errors.Wrap(err, "setting up client")
+	}
+
+	var out *secretsmanager.GetSecretValueOutput
+	var err error
+	msg := awsutil.MakeAPILogMessage("GetSecretValue", in)
+	if err := utility.Retry(
+		ctx,
+		func() (bool, error) {
+			out, err = c.sm.GetSecretValueWithContext(ctx, in)
+			if awsErr, ok := err.(awserr.Error); ok {
+				grip.Debug(message.WrapError(awsErr, msg))
+				switch awsErr.Code() {
+				case request.InvalidParameterErrCode, request.ParamRequiredErrCode:
+					return false, err
+				}
+			}
+			return true, err
+		}, *c.opts.RetryOpts); err != nil {
+		return nil, err
+	}
+	return out, err
```

Comment:
Maybe I missed this, but you can `return out, nil` (which I think I explained in the previous CR).